beesdoo_shift/wizard/validate_attendance_sheet.py

```diff
@@ -0,0 +1,138 @@
+
+    def _get_card_support_setting(self):
+        return (
+            self.env["ir.config_parameter"].get_param(
+                "beesdoo_shift.card_support"
+            )
+            == "True"
+        )
+
+    @api.multi
+    def _get_warning_regular_workers(self):
+        """
+        A warning is shown if some regular workers were not expected
+        but should be doing their regular shifts. This warning is added
+        to sheet's notes at validation.
+        """
+        sheet = self._get_active_sheet()
+        warning_message = ""
+        if sheet:
+            for added_shift in sheet.added_shift_ids:
+                is_regular_worker = added_shift.worker_id.working_mode == "regular"
+                is_compensation = added_shift.is_compensation
+
+                if is_regular_worker and not is_compensation:
+                    warning_message += (
+                        _(
+                            "\n%s attended its shift as a normal one but was not expected. "
+                            "Something may be wrong in his/her personnal informations.\n"
+                        )
+                        % added_shift.worker_id.name
+                    )
+        return warning_message
+
+    @api.multi
+    def _get_default_notes(self):
+        if self._get_active_sheet():
+            return self._get_active_sheet().notes
+
+    @api.multi
+    def _get_default_feedback(self):
+        if self._get_active_sheet():
+            return self._get_active_sheet().feedback
+
+    @api.multi
+    def _get_default_worker_nb_feedback(self):
+        if self._get_active_sheet():
+            return self._get_active_sheet().worker_nb_feedback
+
+    card_support = fields.Boolean(default=_get_card_support_setting)
+    login = fields.Char(string="Login")
+    password = fields.Char(string="Password")
+    barcode = fields.Char(string="Barcode")
+    warning_regular_workers = fields.Text("Warning",
+        default=_get_warning_regular_workers,
+        help="Is any regular worker doing its regular shift as an added one ?"
+    )
+    notes = fields.Text(
+        "Notes about the attendance for Members Office",
+        default=_get_default_notes,
+    )
+    feedback = fields.Text("Comments about the shift", default=_get_default_feedback)
+    worker_nb_feedback = fields.Selection(
+        [
+            ("not_enough", "Not enough"),
+            ("enough", "Enough"),
+            ("too_many", "Too many"),
+        ],
+        string="Was your team big enough?",
+        default=_get_default_worker_nb_feedback,
+        required=True,
+    )
+
+    def on_barcode_scanned(self, barcode):
+        self.barcode = barcode
+
+    @api.multi
+    def save(self):
+        """
+        Save modifications onto attendance sheet.
+        """
+        sheet = self._get_active_sheet()
+
+        sheet.notes = self.notes
+        sheet.feedback = self.feedback
+        sheet.worker_nb_feedback = self.worker_nb_feedback
```

Comment:
if active_sheet was a field, you could persist it directly.

Reply:
Still, as it is a wizard I should not override `write` method, don't you think ?